tests/spec_decode/test_multi_step_worker.py

```diff
@@ -690,3 +691,55 @@ def test_use_draft_model_runner_advance_step():
         worker.execute_model(execute_model_req=execute_model_req)
     call_args_list = worker.model_runner._gpu_advance_step.call_args_list
     assert len(call_args_list) == 1
+
+
+@torch.inference_mode()
+def test_expand_execute_model_request_sync_with_expand_hidden_states():
```

Comment:
Currently, this assumes `HiddenStates` are only used when all seq_groups have 1 sequence each (as per the assertion in `HiddenStates.__init__`). If that assertion needs to be relaxed, we can modify this test to add a more general case.

The way this works is by matching the order of seqs in expanded seq_group_metadata_list with the order of previous_hidden_states in expanded execute_model_request.

Reply:
Nice. What happens if a request is aborted?

Reply:
OK I think there is an issue here where a sequence may be "paused" then "resumed" later. so this should only remove things that are confirmed aborted.

but we can fix that later.

Reply:
can you add a comment for this?

Reply:
Added this comment to `HiddenStates.prune`